### DISCUSSION TITLE
Do any real world applications use Borland's BGI graphics?

### DISCUSSION THREAD

**Discussion by andrewbird** [dosemu2/dosemu2#1372] [General]
I read on the BTTR forum that some previously closed source BGI SVGA drivers have been released under an MIT licence, see https://github.com/jharg93/SvgaBGI. In the distant past (pre Windows era) I used BGI with Turbo Pascal, but I don't think I've ever come across any commercial or other software that uses BGI. Does anyone know of any?

**Comment by stsp** [comment]
I think my volksvga game that can be
found in some tickets here, used svga.bgi
from hargraves.
In any case, you might be kidding by saying
no one uses bgi.
What's the deal anyway?

**Reply by andrewbird** [reply]
Honest question, I don't know of any real world applications that used BGI, no pun or [double entendre]( https://en.wikipedia.org/wiki/Double_entendre). I thought it might be possible to write a bgi to sdl shim, and was just trying to gauge usefulness.

**Comment by stsp** [comment]
Well, many games used BGI, for example
most of all Gamos games before 95, IIRC:
https://ru.wikipedia.org/wiki/Gamos
Even if you don't use BGI drawing primitives,
you could still use it as a portable way
of setting the desired video mode and
drawing gfx fonts.
Also the video modes back then
didn't have the LFB, so drawing
primitives were quite handy too.

> to write a bgi to sdl shim, and was just trying to gauge usefulness.

What's the desire?
Some googling reveals there were
BGI port somewhere for djgpp.

**Comment by stuaxo** [comment]
I guess Borland Quatro Pro is the one that comes up.

It looks like there was a BGI implementation for Windows too.

https://www.cs.colorado.edu/~main/bgi/doc/bgi.html

**Comment by andrewbird** [comment]
> What's the desire?

None really, mainly just curiosity! Sometimes I see things that have just been released under a free licence (like SvgaBGI) and wonder if they can be useful in some way.

**Reply by stsp** [reply]
If only for resurrecting the djgpp bgi port.

**Reply by stuaxo** [reply]
They seem to support quite a few different SVGA adapters, so the code might if any of them were to be implemented.

**Reply by stsp** [reply]
Not sure I understand that sentence,
but if you are talking about the djgpp
port, then it doesn't matter what the
adapters did the original code supported.
All that matters is the same API.
